src/utils/api_helper.ts

```diff
@@ -186,7 +187,7 @@ export const internalFetch = async <T>(
         (e instanceof ApiFetchError && e.status >= 500))
 
     if (isRetryableError) {
-      const retryCount = 3 - maxRetries
+      const retryCount = maxRetries - remainingRetries
```

Comment:
Did you read my comments before writing this?

Reply:
I mentioned that you should remove the retryCount entirely and rely solely on maxRetries. You don’t need to pass an additional variable to control recursion; that’s not how recursive algorithms are intended to work. The only argument needed in this function is withRetry and maxRetries;Also, rename maxRetries to remainingRetries and use it directly within the function.

Right now, your approach causes the first retry to take longer than necessary, which breaks the intended retry timing logic. Simplifying it this way ensures the recursion behaves predictably and the delays scale correctly as intended.